controllers/openstackclient_controller.go

```diff
@@ -385,32 +399,48 @@ func (r *OpenStackClientReconciler) podCreateOrUpdate(instance *ospdirectorv1bet
 	}
 
 	// create k8s.v1.cni.cncf.io/networks network annotation to attach OpenStackClient to networks set in instance.Spec.Networks
-	annotation := "["
-	for id, netName := range instance.Spec.Networks {
+	annotations := ""
+	for id, netNameLower := range instance.Spec.Networks {
+		// get network with name_lower label
+		labelSelector := map[string]string{
+			openstacknet.SubNetNameLabelSelector: netNameLower,
+		}
+
 		// get network with name_lower label
 		network, err := openstacknet.GetOpenStackNetWithLabel(
 			r,
 			instance.Namespace,
-			map[string]string{
-				openstacknet.SubNetNameLabelSelector: netName,
-			},
+			labelSelector,
 		)
 		if err != nil {
 			if k8s_errors.IsNotFound(err) {
-				r.Log.Info(fmt.Sprintf("OpenStackNet with NameLower %s not found!", netName))
+				common.LogForObject(
+					r,
+					fmt.Sprintf("OpenStackNet with NameLower %s not found!", netNameLower),
+					instance,
+				)
 				continue
```

Comment:
Should we make this similar to lines 652-659 in terms of logging the error and updating the status?

Reply:
yes we probably should do that, but we should never see this error here as we already created the ipset where the networks is also checked, but lets add it and not continue, instead return a real error with condition.